homeassistant/components/geniushub/climate.py

```diff
@@ -0,0 +1,204 @@
+"""
+Supports Genius hub to provide climate controls.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/climate.geniushub/
+"""
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    STATE_ECO, STATE_HEAT, STATE_AUTO, STATE_IDLE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_OPERATION_MODE,
+    SUPPORT_ON_OFF, SUPPORT_AWAY_MODE)
+from homeassistant.components.geniushub import GENIUS_HUB
+from homeassistant.const import (
+    ATTR_TEMPERATURE, STATE_OFF, STATE_ON, TEMP_CELSIUS)
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'geniushub'
+
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE \
+    | SUPPORT_ON_OFF | SUPPORT_AWAY_MODE
+# Genius supports the operation modes: Off, Override, Footprint and Timer
+# To work with Alexa these MUST BE
+#
+#   climate.STATE_HEAT: 'HEAT',
+#   climate.STATE_COOL: 'COOL',
+#   climate.STATE_AUTO: 'AUTO',
+#   climate.STATE_ECO: 'ECO',
+#   climate.STATE_IDLE: 'OFF',
+#   climate.STATE_FAN_ONLY: 'OFF',
+#   climate.STATE_DRY: 'OFF',
+
+# These needed to be mapped into HA modes:
+# Off       => OFF      => STATE_IDLE   # Mode_Off: 1,
+# Override  => HEAT     => STATE_HEAT # Mode_Boost: 16,
+# Footprint => ECO      => STATE_ECO    # Mode_Footprint: 4,
+# Timer     => AUTO     => STATE_AUTO   # Mode_Timer: 2,
+# Away                      # Mode_Away: 8,
+#
+OPERATION_LIST = [STATE_IDLE, STATE_HEAT, STATE_ECO, STATE_AUTO]
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the Genius hub climate devices."""
+    genius_hub = hass.data[GENIUS_HUB]
+    await genius_hub.getjson('/zones')
+
+    # Get the zones with a temperature
+    climate_list = genius_hub.getClimateList()
+
+    for zone in climate_list:
+        async_add_entities([GeniusClimate(genius_hub, zone)])
+
+
+class GeniusClimate(ClimateDevice):
+    """Representation of a Genius Hub climate device."""
+
+    def __init__(self, genius_hub, zone):
+        """Initialize the climate device."""
+        GeniusClimate._genius_hub = genius_hub
+        self._name = zone['name']
+        self._device_id = zone['iID']
+        self._current_temperature = zone['current_temperature']
+        self._target_temperature = zone['target_temperature']
+        self._mode = zone['mode']
+        self._is_active = zone['is_active']
+
+    @property
+    def state(self):
```

Comment:
Don't override `state` property. That is the responsibility of the base climate entity class. Implement `current_operation` and `is_on` as needed.